ledger/block/src/helpers/target.rs

```diff
@@ -36,12 +36,12 @@ pub fn block_reward<N: Network>(
     time_since_last_block: i64,
     coinbase_reward: u64,
     transaction_fees: u64,
-) -> u64 {
+) -> Result<u64> {
     // Determine which block reward version to use.
-    match block_height < N::CONSENSUS_V2_HEIGHT {
-        true => block_reward_v1(total_supply, block_time, coinbase_reward, transaction_fees),
-        false => block_reward_v2(total_supply, time_since_last_block, coinbase_reward, transaction_fees),
-    }
+    Ok(match N::CONSENSUS_VERSION(block_height)? as usize {
```

Comment:
Why cast to `usize` instead of directly comparing against `ConsensusVersion::V*`?

Reply:
It is very likely we will have 100+ consensus versions. If branch A applies to the first 50, and branch B applies to the next 50, the cleanest way to implement this is to use `Range`s.

We can't use ranges on enums, only on ints. 

Maybe there is additional syntax sugar we can apply, but match arm logic needs to be const, so that would have to be additional macro magic.

Reply:
New idea: instead or on top of defining consensus versions in an enum, we define a bunch of `const CONSENSUS_V1: usize = 1;`. That may be clearer than a raw `1`. But it doesn't help *that* much and its a line more boilerplate to write for each version. Let me know what you prefer.

Reply:
I think keeping a usize comparison is fine, but would prefer to compare it against `ConsensusVersion::V1` instead of just `1`.

Ranges can be done with: `0..ConsensusVersion::V99` and `ConsensusVersion::V99..`

Reply:
I wrote above: We can't use ranges on enums, only on ints.

Reply:
You should be able to cast it to an integer

Reply:
Unfortunately non-const values or functions are not allowed. Even the `as_int()` is not allowed in a match arm.

However, I realise that we don't use the advantage of using match arms anyway. The compiler could help us and complain whether we're covering all versions, but this doesn't apply because we're using a catch all `_` case for the latest versions to avoid having to update all checks for every consensus version.

Implemented comparison against `ConsensusVersion` using if-else statements in a new commit here: 91018e522